tsdb/head_read.go

```diff
@@ -103,11 +103,18 @@ func (h *headIndexReader) LabelNames(matchers ...*labels.Matcher) ([]string, err
 
 // Postings returns the postings list iterator for the label pairs.
 func (h *headIndexReader) Postings(name string, values ...string) (index.Postings, error) {
-	res := make([]index.Postings, 0, len(values))
-	for _, value := range values {
-		res = append(res, h.head.postings.Get(name, value))
+	switch len(values) {
+	case 0:
+		return index.EmptyPostings(), nil
+	case 1:
+		return h.head.postings.Get(name, values[0]), nil
+	default:
+		res := make([]index.Postings, 0, len(values))
+		for _, value := range values {
+			res = append(res, h.head.postings.Get(name, value))
+		}
+		return index.Merge(res...), nil
```

Comment:
I changed this because `len() == 1` is now (was it before?) a hot path, and I thought it would worth removing the extra unnecessary allocations, does this need a benchmark too?

Reply:
No need of benchmark for this. It is only removing stuff for this case and not changing anything.